assets/js/modules/analytics/components/common/AcquisitionSources.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * DashboardAllTrafficWidget component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getDataTableFromData, TableOverflowContainer } from '../../../../components/data-table';
+import MiniChart from '../../../../components/mini-chart';
+import { numberFormat } from '../../../../util/i18n';
+import { getCurrentDateRange } from '../../../../util/date-range';
+
+function AcquisitionSources( { data, args } ) {
+	if ( ! data ) {
+		return null;
+	}
+
+	const hasTotals = Array.isArray( data[ 0 ].data.totals ) && data[ 0 ].data.totals.length;
+	const hasRows = Array.isArray( data[ 0 ].data.rows ) && data[ 0 ].data.rows.length;
+	if ( ! hasTotals || ! hasRows ) {
+		return null;
+	}
+
+	const options = {
+		hideHeader: true,
+		chartsEnabled: true,
+	};
+
+	let keyColumnIndex = 0;
+	let headers = [
+		{
+			title: __( 'Source', 'google-site-kit' ),
+			primary: true,
+		},
+		{
+			title: __( 'Percent', 'google-site-kit' ),
+		},
+	];
+
+	const { dateRange, url } = args;
+	if ( url ) {
+		const currentRange = getCurrentDateRange( dateRange );
+
+		keyColumnIndex = 1;
+		options.hideHeader = false;
+		options.chartsEnabled = false;
+
+		headers = [
+			{
+				title: __( 'Channel', 'google-site-kit' ),
+				tooltip: __( 'Channel refers to where your traffic originated from', 'google-site-kit' ),
+			},
+			{
+				title: __( 'Users', 'google-site-kit' ),
+				tooltip: __( 'Number of users that originated from that traffic', 'google-site-kit' ),
+			},
+			{
+				title: __( 'New Users', 'google-site-kit' ),
+				/* translators: %s: date range */
+				tooltip: sprintf( __( 'Number of new users to visit your page over last %s', 'google-site-kit' ), currentRange ),
+			},
+			{
+				title: __( 'Sessions', 'google-site-kit' ),
+				/* translators: %s: date range */
+				tooltip: sprintf( __( 'Number of sessions users had on your website over last %s', 'google-site-kit' ), currentRange ),
+			},
+			{
+				title: __( 'Percentage', 'google-site-kit' ),
+				tooltip: __( 'Percentage of sessions', 'google-site-kit' ),
+			},
+		];
+	}
```

Comment:
This logic somehow doesn't seem to work. When I look at the widget in the single URL details view, it still only shows the source and percentages like in the overall dashboard, not the five columns that it should as defined in the code here.

Reply:
Hm... @felixarntz it works correctly on my end. Perhaps you forgot to re-build assets after pulling my changes? :slightly_smiling_face: Can you re-build assets and check again? 

See the storybook for this PR:

#### Dashboard
![dashboard](https://user-images.githubusercontent.com/873049/91667169-1a803980-eb0b-11ea-8969-6955247d0d80.png)
https://google.github.io/site-kit-wp/storybook/pull/1931/?path=/story/analytics-module-components-dashboard-all-traffic-widget--loaded

#### Page Dashboard
![page-dashboard](https://user-images.githubusercontent.com/873049/91667127-df7e0600-eb0a-11ea-9708-b652596738f4.png)
https://google.github.io/site-kit-wp/storybook/pull/1931/?path=/story/analytics-module-components-page-dashboard-all-traffic-widget--loaded

Reply:
@eugene-manuilov It works on Storybook indeed, but not on the actual details view page, so there's must be something off contextually there.